apps/src/gamelab/GameLabP5.js

```diff
@@ -125,6 +125,41 @@ GameLabP5.prototype.init = function (options) {
     return false;
   };
 
+  // Modify p5 to ignore out-of-bounds positions before setting touchIsDown
+  window.p5.prototype._ontouchstart = function (e) {
+    if (!this._curElement) {
+      return;
+    }
+    var validTouch;
+    for (var i = 0; i < e.touches.length; i++) {
+      validTouch = getTouchInfo(this._curElement.elt, e, i);
+      if (validTouch) {
+        break;
+      }
+    }
+    if (!validTouch) {
+      // No in-bounds (valid) touches, return and ignore:
+      return;
+    }
+    var context = this._isGlobal ? window : this;
+    var executeDefault;
+    this._updateNextTouchCoords(e);
+    this._updateNextMouseCoords(e);
+    this._setProperty('touchIsDown', true);
+    if (typeof context.touchStarted === 'function') {
+      executeDefault = context.touchStarted(e);
+      if (executeDefault === false) {
+        e.preventDefault();
+      }
+    } else if (typeof context.mousePressed === 'function') {
+      executeDefault = context.mousePressed(e);
+      if (executeDefault === false) {
+        e.preventDefault();
+      }
+      //this._setMouseButton(e);
```

Comment:
Is there a reason this line is commented out?

Reply:
That's the way it is in the original `p5` source. I wanted to leave it that way to ease our future diffs.